### DISCUSSION TITLE
USING ASK.ASK() TO SUCCESSFULLY INTERACT WITH CONTRACT

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [reach-sh/reach-lang#1109] [Q&A] [answered]
Hello Reach Fam,

I am trying to interact with a deployed Reach contract using the Reach Frontend and there is a strange error I am getting, this is my code:

```interact.provideAssetDetails = async () => {
          /*
          assetName: Bytes(58),
          assetType: Bytes(58),
          assetValue: UInt,
          ownerAddress: Address
          */
          const assetObj = new Object();

          /*
          ask.ask(string, (string => result)) => Promise<result>
          ask.yesno(string) => boolean
          ask.done() => null
          */
          const assetName = await ask.ask(
            `Enter the asset name:`,
            (x => x)
          );
```

The error at the console is:

```
Starting dApp...
TypeError: ask.ask is not a function
 at Object.interact.provideAssetDetails (file:///workspace/mydApp/indexApi.mjs:133:33)
    at Tokenizer (file:///workspace/mydApp/build/index.main.mjs:2454:53)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async tokenizerOrAdminInteractions (file:///workspace/mydApp/indexApi.mjs:176:7)
    at async indexApi (file:///workspace/mydApp/indexApi.mjs:270:31)
Done in 4.40s.

```

This error is strange because looking at the Reach documentation, it is clear that ```ask.ask()``` is an asynchronous function and I treated it as one in my code, and quote:

```
[ask](https://docs.reach.sh/frontend/#js_ask)[.ask](https://docs.reach.sh/frontend/#js_ask.ask) is an asynchronous function that asks a question on the console and returns a Promise for the first result that its second argument does not error on.[181](https://docs.reach.sh/frontend/#p_181)

```
Link is here: [https://docs.reach.sh/frontend/#ref-frontends-js-ask] 

What could be the issue?

**Comment by jeapostrophe** [comment, marked as answer]
Did you import it? https://docs.reach.sh/frontend/#p_178

**Comment by kcelestinomaria** [comment]
There seems to be an issue, I imported it this way: ```import { ask, done } from '@reach-sh/stdlib';``` which is what produces the error since when I solely import ask i.e ```import { ask } from '@reach-sh/stdlib';``` , but I wanted to use done() to close the main function in my application, is this possible or the feature deprecated

**Reply by jeapostrophe** [reply]
The documentation I linked has everything you need to know --- https://docs.reach.sh/frontend/#ref-frontends-js-ask--- you are looking for `ask.done`, just like the documentation says. There has never been a `done` export from `stdlib`
